crates/daphne-server/src/router/mod.rs

```diff
@@ -246,33 +273,9 @@ where
             .map_err(|e| (StatusCode::BAD_REQUEST, e.to_string()))?;
 
         let extract_header_as_string = |header: &'static str| -> Option<String> {
-            parts
-                .headers
-                .get(header)?
-                .to_str()
-                .ok()
-                .map(ToString::to_string)
+            extract_header_as_string(&parts.headers, header)
         };
```

Comment:
Let's just completely get rid of this closure and inline `extract_header_as_string`.